### ISSUE TITLE
ML/DAY counter

### ISSUE THREAD

**Issue by falconettt** [ClockSelect/myevic#51]
Yes, it works. Great. Thanks, thanks. THANKS! -))
But IMHO some bug at the end. 
Mod gives "Long fire protection" and display freeze at this phrase, you need to press fire to return to main screen. 
I think, better no freeze, pause 3-5 sec and automatic return to main screen.

Sorry for my english.

**Comment by Daveee10** [comment]
That's not bug. The 10s protection works seems like this, only the time now editable.
IMO not bad how the "warning" and "error" messages stay on screen while you don't press a button. IMO....

**Comment by falconettt** [comment]
ClockSelect,
I insist on my logic. Because my logic is not for protection, but to limit vaping. 
I vape too much, it's not good. 
If you want protection only just live it 10 sec.

**Comment by Daveee10** [comment]
Maybe for you a better feature is the puff limit. I think what on SMOK mods. You can set a max puff number and when you reaches that you can't vape, just if you reset the counter or extend the limit... I thinking on this feature. Maybe ClockSelect will tell his opinion or make his modificaton on the long fire protection. Maybe :)

**Comment by falconettt** [comment]
IMHO puff limit is not convenient. Better some statistical number (some math combination of time and puff). Let's call it "velocity of vaping". If number is low - you vape. If number is high - you do not. And you decide what is high and what is low.  

Other choice:
The speed with which you drain the battery is "velocity of vaping".
JUST SHOW IT ON MAIN SCREEN!

**Comment by ClockSelect** [comment]
What do you mean by "vaping too much"?
Sounds like a nonsense to me.

**Comment by falconettt** [comment]
I'll explain the SENSE. 
There is a need to control the amount of evaporated liquid. This somehow implemented in all advanced devices. First, the length of inhaling (you almost did). Secondly, how often do you do these inhaling (speed). How it can be implemented? For example, like this: 
http://www.joyetech.com/product/control-head/
![123](https://cloud.githubusercontent.com/assets/22370680/18829358/923e6038-83e4-11e6-8409-07a25940ff0d.jpg)

Look at smile icon.

**Comment by ClockSelect** [comment]
Yes I think I see what you want.
Maybe it could be implemented as an advancement bar on the third line like:
VVEL [|||||||    ]
Telling you how far you are from your limit.
Option would be available only if you configure a limit.
Quantity of evaporated liquid is a time integral of the applied power, have to made some research to approximate it correctly.
A suitable limit would be in mL/hour, for example? I think the number of puffs has no incidence, really - it doesn't tell you how _much_ you vape at all.
That needs discussion.

**Comment by falconettt** [comment]
Я вижу мы наконец поняли друг друга. Это радует.-)). 
I see we finally understand each other. This is good.-)).
Yes, time integral of the applied power best represents the speed of consumption.
IMO: there is no need to establish a physical limit and block vaping, just show the speed figure on main screen or even better on screen saver.
I'll test your invention and you will decide later what to do next. Okay?
ML/DAY IS IMPORTANT!!!!     NOT    ML/HOUR!!! 
Because one hour you can be patient and do not vape, then the velocity is zero, but that means nothing.

Yes, the number of puffs doesn't tell you how much you vape at all.

**Comment by ClockSelect** [comment]
The main question is how to make it reliable.
Displaying a useful information with 64 pixels and when and where is another issue.
Thinking about it.

**Comment by ClockSelect** [comment]
First rough estimation: a full 4.5ml tank took around 10 minutes to empty at 30W.
Didn't truly realize before testing^^
That's 250nL/s/W

**Comment by falconettt** [comment]
Look here:
https://en.wikipedia.org/wiki/Joule

OK, full 4.5ml tank took around 10 minutes to empty at 30W

10 min = 600 sec

600s x 30w = 18000 J

You have vaping time in time counter.
You have day time because you have clock.

Suppose now 16 o'clock. It's 57600 seconds from the beginning of the day.

VVEL = 18000 J / 57600 s = 0,3125

Suppose now 17 o'clock. It's 61200 seconds from the beginning of the day.

VVEL = 18000 J / 61200 s = 0,2941

Suppose now 18 o'clock. It's 64800 seconds from the beginning of the day.

VVEL = 18000 J / 64800 s = 0,2777

You see, VVEL is quite reliable this way (If the wattage does not change during the day, but if change it is necessary to calculate the weighted average)

**Comment by ClockSelect** [comment]
Yes but at 1 o'clock VVEL = 7. What would it mean?
Measurement should be made in a sliding time window.

**Comment by falconettt** [comment]
Suppose now 1 o'clock. It's 3600 seconds from the beginning of the day:

If you do not vape at all:
VVEL = 0 J / 3600 s = 0

If you vaped full 4,5 ml:
VVEL = 18000 J / 3600 s = 5  (you see 5 and, WOW, you understand that it is time to slow down)

If you make "sliding time window" - even better, but let it be as long as possible.
I want to know my VVEL for the week.

And wattage change during the period, it is necessary to calculate the weighted average.

**Comment by falconettt** [comment]
May be better this way?
VVEL = 0 J / 60 min = 0
VVEL = 18000 J / 960 min = 18,7
VVEL = 18000 J / 1020 min = 17,6
VVEL = 18000 J / 1080 min = 16,7
VVEL = 18000 J / 60 min = 300
VVEL = 18000 J / 10 min = 1800

We have a range of variation VVEL from 0 to 1800.

What would it mean?
Suppose you want to vape not more than 10 ml a day.
10ml needs 22 min of vaping time and 22x60x30=39600 J of energy.
24 hour = 1440 min.
VVEL=39600 J / 1440 min = 27,5

Just keep your VVEL not more than 27,5 and you get not more than 10 ml of liquid a day!

JUST DO IT!   -))

**Comment by falconettt** [comment]
You'd probably have to implement it some other way.
But think of it as the speedometer in the car.
If you want to drive 60 kilometers  in one hour you have to keep an average speed of 60 km/h.
I hope this is clear...

**Comment by ClockSelect** [comment]
I made an experimental VVel on the third line.
At the moment, it won't resist a battery swap.

**Comment by falconettt** [comment]
GREAT!
I'm testing...
First impression - that's what I wanted to get!
I do not swap batteries but charge via USB.

Can you give some instructions? 

Can I change atoms and wattage?
How to reset and start from the beginning?
Does VVEL starts again automatically  from the begining at 00:00 o'clock?
Does it need tarification?
And so on  ...

Definitely need to put VVEL on screensaver.
What is the accuracy?
Even if the accuracy is not good and I get 12 instead of 10 milliliters of the desired. The next day I will try to get 8 on the screen and I get 10 in reality.

**Comment by ClockSelect** [comment]
Atm, calculation is made from box startup.
One may reset value by reseting the box. There is no automatic reset.
You can change ato & wattage as you want. Should also work in temp control, or any other mode.
I have no idea of the accuracy, it needs more testing; but it shouldn't be completely off.
What do you mean by "tarification" ?

**Comment by falconettt** [comment]
tarification:
Full 4.5ml tank took around 10 minutes to empty at 30W  IS NOT the rule.
It's all different on different atoms and in different circumstances.
Additionally, you may experience inaccuracy of changes in the ambient temperature.

Can I reset time and puff or it will spoil the measurement of vvel?

Definitely need:
1. Put it in readme, we need more testers.
1. Manual reset any time you want.
2. Automatic reset every day (if you want) and at what time you want.
3. Put VVEL on screensaver.

OК, let's test as it is for a few days...

**Comment by falconettt** [comment]
2 hour test result:
How to reset the box properly?
I've uploaded soft once more (I hope this is  also reset) and start.
I kept vvel = 24 ml/d on the screen for 2 hours. It is 1ml/hour velocity.

2 ml of liquid vaped in 105 min instead of 120 min. Accuracy is 87%. Not bad.

I'll start 24 hour test now, you'll get the result tomorrow.

**Comment by falconettt** [comment]
Good news!

24 hour test:
Uploaded soft to reset and started.
I didn't change anything during the test. (atom, wattage, air temperature and so on).

![vvel-test](https://cloud.githubusercontent.com/assets/22370680/19004748/f369a68c-875f-11e6-8d81-43d4ceea8972.jpg)

You see, it has accuracy about 80%. And accuracy doesn't change during the period of test!

Just simple input for linear compensation will make VVEL accuracy perfect.
IMHO: Definitely needs an automatic reset each day and manuall reset if necessary.

If you go outside in winter or start to change something dramatically - may be the result'd be worse ... 
But I usually don't.

What's next?

**Comment by ClockSelect** [comment]
What was your test config? (coil type, power, etc)
I'd like to know if there's significant differences between setups.
Should be tested in TC mode, too.

If you want to reset your box, there should be a pinhole on the bottom to access the reset button with a needle or an unfolded paperclip (it's the case on the VTwo, don't know for other boxes). Anyway I'll add a reset function.

**Comment by falconettt** [comment]
Test config was:

evic vtc mini
harware version 1.01
battery sony vts4 in GEN mode.
usb charge and passthrough.
contrast 0%
Stealth ON.
SS 316L / 0,4 om / 18-19 w
VW without preheat.

**Comment by falconettt** [comment]
You need MORE TESTERS.
Can try Kanthal and TC mode, but I can't try 50w cause I'll die at this wattage ... -)

**Comment by falconettt** [comment]
TCR 120 / SS 316L / 0,28 om / 20-22 watt / 220-240 C.

![vvel2](https://cloud.githubusercontent.com/assets/22370680/19016148/fa36eb08-881a-11e6-91fe-52c33d12e253.jpg)

You see, the result is just the same.

I'll try kanthal on monday.

**Comment by falconettt** [comment]
Bad news,
Uploaded new soft to reset and started.
I noticed you made some changes...
VW with preheat, kanthal, 2 coils, 0,65 om, 26,5 watt.
![3](https://cloud.githubusercontent.com/assets/22370680/19065345/ac842670-8a1b-11e6-8b74-f62c21350fbf.jpg)

Reseted VVEL and started again.
The result is the same.
![4](https://cloud.githubusercontent.com/assets/22370680/19065552/d8268312-8a1c-11e6-9343-ae3710cc803d.jpg)

What's next?

**Comment by ClockSelect** [comment]
I did not make changes (well, I mean wanted changes), but I noticed something wrong too.
Investigating.

High wattage is not the point; people vaping at 60+ watts do not care about their consumption.

TC is more relevant; I think it's where we should be the more accurate.

**Comment by falconettt** [comment]
Stopped testing. Please let me know when you find the solution. What test should I begin?

**Comment by PaoloW8** [comment]
ClockSelect, I can tell you that I noticed that VVEL stopped working after this commit
https://github.com/ClockSelect/myevic/commit/30b833169eccfdb3c3a927ef816df8e33386574d

I hope this information can help you in some way

**Comment by Daveee10** [comment]
Maybe this line cause the problem in mainview.c:
vv /= 10;

**Comment by ClockSelect** [comment]
Right: didn't "move" the vv / 10, duplicated it.
So valus should just by multiplied by 10; does not really prevent testing^^

**Comment by ClockSelect** [comment]
So I changed a few things, beyond fixing the divide-by-ten issue:
- When you reset the ml/d counter, the reference date is set to today midnight; so best time to reset counter is in the morning before starting vaping. Obviously, the counter will not be accurate before a full day passed. It should remain accurate after that, and will show you your average consumption over the period. It's a slowly-evolving counter.
- Counter is now memorized and should survive resets and battery swaps (but since it uses RTC hardware, remember battery must be changed fast on non-RTC enabled devices).

I'm planing on an option to display just total milliliters vaped since last counter reset, just like puff and time. Will make tests simpler and also work without having to setup the clock.

**Comment by falconettt** [comment]
IMHO:
No matter what is the accuracy by itself. The main thing that the accuracy obviously will change (more or less) during the  period of measurement. The longer the measurement period the greater the error. And accuracy will be different for different mods and for different temperature and so on. This counter will slowly accumulate error. 
- At this time, I don't need the period to be longer than 24 hour. What for? 
- I want to be able to carry out linear correction, just some number input for real velocity.
- Different reference time will give different experience, may be better reset time as reference, or let me input.

Understand that your tasks, perhaps much wider...

Sorry for my english again, lets test!

**Comment by ClockSelect** [comment]
I've added the option to display total mL instead of mL/d; some people may prefer this option. What's more, it will permit to judge the accuracy of liquid measurement.

@falconettt : A Watt is a Watt; there's no reason to have high discrepancies between setups. I'm curious we measure that; can people give us their measurements of mL in diffrent setups? like temperature control? Reset the counter, fill a tank of known capacity, vape the tank and check.
The value set up for consumption is 320nL/J.

For the period: the best to do atm is to reset your counter when waking up in the morning. Not a big deal. For configurable reference time and corrective factor, I don't want to make complicated things, menu etc, for what remains a secondary feature imo. But future will tell.

Maybe another time than midnight will be preferable, such as something between 3~6:00, so that accuracy comes faster during the start of the day. Don't know, should test.

**Comment by ClockSelect** [comment]
I upped consumption ratio to 360mL/kJ, this fits my setup, so it should be a good starting base for tests.

**Comment by falconettt** [comment]
PERFECT!
Quick test 1 ml:
VW with preheat, kanthal, 2 coils, 0,65 om, 26,5 watt
1 ml in 62 min.
started 20:10,   ended 21:12
its 1272 min from the beginig of the period.
1/1272*1440=1,13 ml/day
vvel = 1,13 ml/day
accuracy 100%
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
It was before you upped consumption ratio to 360mL/kJ

what's next?

**Comment by ClockSelect** [comment]
Your correct reading will be at midnight. But since I upped the ratio by 25% and you read at around 75% of the period, you've just been lucky.
I encourage you to check raw mL ATM, not mL per day. We will handle velocity in a later time.

**Comment by falconettt** [comment]
Let's make it this way:

We have two modes: "test mode" in ML and "measure mode" in ML/DAY.
You start in "test mode", reset it to zero and vape 1ml. 
You see on the screen some digit other than 1.
You adjust  with (+/-) buttons  this  digit to be 1ML ( adjusting consumption ratio actually, without even knowing what is it).
You switch to "measure mode" and get your pleasure with accuracy 100%.

Yes?

**Comment by falconettt** [comment]
02:00   Uploaded soft with  consumption ratio 360mL/kJ,
11:00   Played with and reseted ML mode.
12:58   Reseted ML mode and started 1 ml (SS 316L / 0,37 om / 20,8 w/ VW with preheat 170%/0,3s.)
13:40   Ended test. 1 ml in reality, 1,03 on the screen.

Accuracy 1,03/1*100%=103%. 
3% difference can be neglected. 
So, great again !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!

**Comment by falconettt** [comment]
18:21  reseted, started 1ml Nichrome Ni80/1,09om/16,9-18w/VW with preheat 170%/0,3s.
19:40 ended with 1,09

accuracy 1,09/1*100%=109%

Not bad ..........................

**Comment by falconettt** [comment]
Let's make it this way:

We have two modes: "test mode" in ML and "measure mode" in ML/DAY.
You start in "test mode", reset it to zero and vape 1ml. 
You see on the screen some digit other than 1.
You adjust with (+/-) buttons this digit to be 1ML ( adjusting consumption ratio actually, without even knowing what is it).
You switch to "measure mode" and get your pleasure with accuracy 100%.

Yes?

**Comment by falconettt** [comment]
SS316L/0,38om/20,9w/VW with preheat 170%/0,2s
2ml, 2,09ml on the screen.

accuracy 2,09/2*100%=105%

It's time to let me correct consumption ratio.

**Comment by ClockSelect** [comment]
Now you can adjust the vaping ratio in mL/kJ; not the most practical since you have to do the maths, but eh.

**Comment by falconettt** [comment]
I do 24 hour test now and everything is going great for now. I'll be done in 6 hours and tell you the result.

Please think of ambinent temperature. Should we validate its impact? I mean you make "test mode" at +20 C, and then go outside to vape whith +5 C.

**Comment by falconettt** [comment]
24 hour test:
I don't think there is a need to discuss the accuracy. It's on the side of the user and "test mode" now. We've got a completely accurate result.
I can say that the 24 hour result is about 96-104% and it's GREAT!
That means if you want to get 10ml/day you'll get about 9,6-10,4 ml/day.

Is's all about 20-40 C temperature on the board.

What about to configure reference time and automatic reset every day?

**Comment by ClockSelect** [comment]
I'll have to do a whole sub-menu just for the vvel^^

**Comment by Daveee10** [comment]
Want to make some configurable thing for VVEL or why? Just asking.

Other I don't want create other issue for this idea:
Maybe the Game is a complete menu instead of sub-menu of screen because this is not screen setting.
Another: I will happy if you make some other date format option seems like YYYY-mm-dd because we use that... (I haven't got idea for the simpliest way)
Third: The original VTwo Mini style analog clock on full screen (Maybe you have the code for that from decompiled VTwo (mini) firmware) :)

**Comment by falconettt** [comment]
If you don't want - I won't insist (everyone has the right to do what he wants)^^
Maybe it's right to keep things as simple as possible...
Maybe later you'll try to improve it.
I like VVEL as it is, and thanks for your work, by the way. 

If you forget to reset VVEL manually every day it becomes unreliable sooner or later. We have not verified the accuracy over a very long period. We understand this. But some people understand nothing, they will complain.

If you want to keep VVEL _"experimental"_ - tell me what to experiment further...

**Comment by ClockSelect** [comment]
There's no reason VVEL would become inaccurate along the days, as long as you keep the clock on time.
But it will give you your average per day for the whole period, not for the current day.

**Comment by falconettt** [comment]
OK. What will be with accuracy if I start to change batteries over a long period? 
Suppose old and new ones and from different manufactures?

**Comment by ClockSelect** [comment]
Should not change anything.

**Comment by falconettt** [comment]
Uploaded _"bc74e95"_ in RX2/3 . Still alive ...^^
ML counter works but ML/DAY COUNTER DOESN'T. Just 00:00 ml/d. 
please fix this

Evic mini : both vvel counters work OK.

**Comment by Daveee10** [comment]
Set the date and time...

**Comment by falconettt** [comment]
already did, doesn't help

**Comment by Daveee10** [comment]
That's weird... ml/day calculated from epoch time and ml... So that's need to work...

**Comment by PaoloW8** [comment]
you have already tried to manually reset the ml/d counter ?
maybe after the reset will start

**Comment by ClockSelect** [comment]
on the rx2/3: I've set up date/time, reset the counter, and the ml and ml/d started to work fine.

**Comment by falconettt** [comment]
Yes, started, thanks

**Comment by Filo301** [comment]
How to reset vvel? i mean total vaped liquid.

**Comment by falconettt** [comment]
3 click fire, 1 click -, fire long click, and you see zero

**Comment by Filo301** [comment]
Thank You!

**Comment by DeDiee** [comment]
Hello! Sorry for my english. I began testing VVEL on CUBOID. Resistance 0.17, power 80W. With the installation of 360 ml / kj accuracy is 95-97%, continue to observe.
The only negative that does not allow the test for a long time, it is reset ml when I change the battery. I try to do it fast, but it does not always work. What can be done?

**Comment by ClockSelect** [comment]
Nothing much atm.
A simple workaround would be to change batteries while plugged on USB (disable battery charging in Expert menu if you do that so that it's safe for the box).

**Comment by DeDiee** [comment]
Thank you!
If it's safe, I'll try it. To begin the test.
Another question: calculation of the power comes from task or from actual power. For example, set 100W, the battery is low, and in fact is 50W.

**Comment by ClockSelect** [comment]
Power setting does not take battery into account; it's only limited by atomizer resistance.
Then when firing, the battery enforces its rules and the displayed power may be lower depending on the state of charge.

**Comment by MoDuLah** [comment]
Don't know why but ml/d on both of my mods (VTC Mini & Cuboid) is stuck on 0.00ml/d

**Comment by Daveee10** [comment]
Set the date and time, then reset the value with long click seems puff counter...

**Comment by MoDuLah** [comment]
Thanks Daveee10 Seems to be rolling again.
Would be awesome to keep the date/time on Cuboid when swapping batteries.
I think it doesn't reset the date/time on VTC..

**Comment by bluenazgul** [comment]
Plug your Box on USB while change the Batteries ;)  it is a little Trick since the Cuboid dont has an real RTC

**Comment by MoDuLah** [comment]
True didn't though of that ;) just wishing to have a usb cable handy at any moment :P damn 80w-100w consume batteries for fun.

**Comment by falconettt** [comment]
To stay tuned I need to update firmware frequently. Does it affect VVEL?
Seems not, but I'm not sure.

You keep label "help wanted". What kind of help you need at this moment?
May be good to point it out in the name of the issue?

**Comment by ClockSelect** [comment]
Help was wanted at some time to calibrate the consumption ratio, didn't remove the label.

**Comment by DeDiee** [comment]
My tests showed that it is necessary to set the parameter ml / kj to 355, so it turns out more precisely.
Can you show the flow rate display in standby, charging etc.?

**Comment by ClockSelect** [comment]
The ratio depens to some extend on the setup, it has to be adjusted. 360 is a good starting point for tuning.
The screen is off in standby; it can hardly display something.

**Comment by DeDiee** [comment]
Yes, I understood and corrected for myself.
I mean, the display fluid under the clockor in place of them when the display work in a charging mode or screensaver mode.

**Comment by Villista** [comment]
I set the value of ml / kJ 365 for a more exact match to "Kanger TOPTANK Mini". But I have three tanks, and more powerful (SMOK TFV8 Cloud Beast) values are not accurate.

**Comment by falconettt** [comment]
Yes, consumption ratio is different for different setups.

**Comment by Villista** [comment]
The ideal solution would be to create profiles (at least three), which contained the:
  mode (power or thermal control) + power level + preheating and delays + mL / kJ
And to give an opportunity to establish a circular shift in the profile of 2/3/4 clicks Fire!

**Comment by falconettt** [comment]
@Villista : There are a lot  "ideal solutions" here. But it needs menu. Owner doesn't won't to develop menu for this moment. You can try to convince him but I failed.

**Comment by SEEKNDESTROY** [comment]
I really don't see the point in having a feature that's never going to be 100% accurate. I don't need a display/device telling me how much I vaped... all I have to do is look at my bottle. I start with a 30ml bottle every morning, by the end of the day it's either empty or close to, and sometimes I'm a little over 30 ml. 30x7=210ml (±) a few ml a week.

**Comment by falconettt** [comment]
If you don't see the point - don't use it. But suppose one having 5 atom and 10 of dark glass bottles in work at the same time. At the end of the day you do not understand at all how much evaporated.

**Comment by enricorov** [comment]
Eventually, I think the dev will introduce some kind of table, or
calibration feature to finetune this parameter.

For now, however, that's what we have.

Il gio 10 nov 2016, 16:42 SEEKNDESTROY notifications@github.com ha
scritto:

**Comment by SEEKNDESTROY** [comment]
What's the purpose of seeing how much you vaped on a display? Is it to tell you that you reached your daily limit? Sorry, I just don't get the purpose of it and that's why I'm not messing with it. Same goes for the game, puff counter and puff time... it doesn't add anything to my vape experience. I guess I'm just a different vaper... for me it's all about improving my vape experience and not about gimmicks that add nothing to it. Maybe I'm the only one that feels that way?

**Comment by falconettt** [comment]
Well, I'm not your mom... ^^   but 30ml a day obviously too much!
What is the purpose of seeing how much you vaped?
The same as odometer and speedometer in the car. Try to use it without...

**Comment by SEEKNDESTROY** [comment]
No you're not my mom, because she's no longer on this earth. 30ml is really not that much at 1.5 mg with the type of RDTA and build I run. I have customers that vape a lot more. I rather vape 30ml a day than smoke a single cigarette... I've been cigarette free for 3 years and have vaped 0% nic for about 8 months. The only reason why I went back on nicotine was because I was having way too many snacks, especially at night. 

You can drive without an odometer and speedometer just fine... you may end up breaking the speed limit but that doesn't affect the way the car drives/handles. The same goes for the ML/DAY counter, it does not change the way your device/setup produces vapor.

Anyway, I posted my 2 cents... in the end it's up to the developer to decide what he wants to do and what's possible to do with the limited amount of space he has to play with.

Let's all have a vape and agree to disagree. :)
Cheers!

**Comment by falconettt** [comment]
> What's the purpose of seeing how much you vaped on a display? Is it to tell you that you reached your daily limit? Sorry, I just don't get the purpose of it and that's why I'm not messing with it. Same goes for the game, puff counter and puff time... it doesn't add anything to my vape experience. I guess I'm just a different vaper... for me it's all about improving my vape experience and not about gimmicks that add nothing to it. Maybe I'm the only one that feels that way?

@SEEKNDESTROY : You can find answer here
http://mentalhealthdaily.com/2015/05/26/hidden-benefits-of-nicotine-on-the-brain/
or you can "google" yourself if interested.

In short, adjusting the amount and speed of nicotine consumption you regulate:
- Alerting attention
- Fine motor skills
- Memory
- Orienting attention
- Response accuracy
- Response time

**Comment by FuckThisAlias** [comment]
does anyone know to get the ml/d to not show 0 mine starts work after about 2 days and thats about when i need to charge the batteries and it resets it back to 0 for 2 days
